doc/HttpTransportAdapter.md

```diff
@@ -0,0 +1,118 @@
+
+## Using the HTTP transport adapter
+The HTTP transport adapter is set up when a service SDK client is init. For example, when creating an Azure Storage SDK client. The HTTP transport adapter is part of the client options argument that can be used to init the client. See the next example.
+```cpp
+  /*
+  * Using the HTTP transport adapter.
+  */
+
+  // When no options are provided, the HTTP transport adapter is set to the default one.
+  auto defaultStorageClient = BlobServiceClient(url, credential);
+
+  // Use client options to set an HTTP transport adapter.
+  BlobClientOptions options;
+  options.TransportPolicyOptions.Transport = std::make_shared<Azure::Core::Http::CurlTransport>();
+  auto storageClient = BlobServiceClient(url, credential, options);
+```
+
+### Re-use the HTTP transport adapter 
+Note that the HTTP transport adapter is a `shared_ptr`. This is because you can re-use the same HTTP transport adapter for multiple clients. There are two ways of doing this. The first one is by using the same HTTP transport adapter to init a second client. The second one is when the client can create a new client directly witch inherits the same configuration. See the next example:
+
+```cpp
+  /* 
+  *  Option 1. Init a new client with the same HTTP transport adapter.
+  */
+  auto curlTransportAdapter = std::make_shared<Azure::Core::Http::CurlTransport>();
+  BlobClientOptions optionsA;
+  optionsA.TransportPolicyOptions.Transport = curlTransportAdapter;
+  auto storageClientA = BlobServiceClient(url, credential, options);
+  // The second client.
+  BlobClientOptions optionsB;
+  optionsB.TransportPolicyOptions.Transport = curlTransportAdapter;
+  auto storageClientB = BlobServiceClient(url, credential, options);
+
+  /* 
+  *  Option 2. Create new client from a parent client.
+  */
+  auto curlTransportAdapter = std::make_shared<Azure::Core::Http::CurlTransport>();
+  BlobClientOptions optionsA;
+  optionsA.TransportPolicyOptions.Transport = curlTransportAdapter;
+  auto storageClientA = BlobServiceClient(url, credential, options);
+  // Create clientB from clientA.
+  auto clientB = storageClientA.GetBlobContainerClient(containerName)
+```
+
+Note that the second option could be not supported for some Azure SDK clients if it doesn't make sense for the service behavior (like Key Vault client).
+
+The Azure SDK for C++ client `will hold an internal copy of the options` passed in the arguments. This means that the options object used to init the client can be disposed. However, since the HTTP transport adapter is a shared ptr, it won't be disposed until any client using it is disposed first.
+
+## HTTP Transport Adapter Options
+The libcurl and WinHTTP transport adapters can also be init with specif options that control features from the HTTP stack underneath it.
+For example, you can set a PROXY or override the default CA path for libcurl. See next example.
```

Comment:
Avoid separating a single paragraph into a new line explicitly, since it has no impact on how the markdown is rendered. Suggested a change to the wording and merge into line 71 above.
```suggestion
```